metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -315,6 +318,12 @@ def build_service_principal_map(self) -> None:
                 "service-principals", f"Unable to fetch service principals: {e}"
             )
 
+    def build_groups_map(self) -> None:
+        try:
+            self.groups += self.unity_catalog_api_proxy.groups()
+        except Exception as e:
+            self.report.report_warning("groups", f"Unable to fetch groups: {e}")
```

Comment:
Nice ! 

As a result of this graceful handling, I don't see any harm in getting this PR in.